apis/Google.Cloud.Spanner.Data/Google.Cloud.Spanner.Data.YCSB/Program.cs

```diff
@@ -0,0 +1,513 @@
+            }
+        }
+
+        private static T GetOptionWithDefault<T>(string optionName, T defaultValue)
+            => !s_options.ContainsKey(optionName) ? defaultValue : GetOption<T>(optionName);
+
+        private static T GetOption<T>(string optionName)
+        {
+            if (!s_options.ContainsKey(optionName))
+            {
+                throw new InvalidOperationException($"Argument {optionName} was not specified, but required.");
+            }
+            var value = s_options[optionName];
+            if (value is T)
+            {
+                return (T) value;
+            }
+            var parsedValue = (T) Convert.ChangeType(value, typeof(T));
+            s_options[optionName] = parsedValue;
+            return parsedValue;
+        }
+
+        private static bool TryParseArguments(IReadOnlyList<string> args)
+        {
+            if (args == null || args.Count == 0 || args[0] != "run")
+            {
+                PrintUsage();
+                return false;
+            }
+
+            if (args[1] != "-P" || args.Count < 3)
+            {
+                PrintUsage();
+                return false;
+            }
+
+            s_templateFile = args[2];
+            if (!File.Exists(s_templateFile))
+            {
+                Console.Error.WriteLine($"File {s_templateFile} does not exist!");
+                PrintUsage();
+                return false;
+            }
+
+            if (args.Count % 2 != 1)
+            {
+                //wierd error, user had a -p without another value or had an extra space in a value causing it
+                //to be misread.
+                Console.Error.WriteLine("Error parsing argument values.");
+                PrintUsage();
+                return false;
+            }
+
+            for (var i = 3; i < args.Count; i += 2)
+            {
+                if (args[i] != "-p")
+                {
+                    Console.Error.WriteLine($"Expected a '-p' at argument {(i - 1) / 2}");
+                    PrintUsage();
+                    return false;
+                }
+                var keyValue = args[i + 1].Split('=');
+                if (keyValue.Length != 2)
+                {
+                    Console.Error.WriteLine($"Unable to parse key/value for arg {(i - 1) / 2}");
+                    PrintUsage();
+                    return false;
+                }
+                s_options[keyValue[0].ToLowerInvariant()] = keyValue[1];
+            }
+            s_debugMode = s_options.ContainsKey("debugmode");
+            return true;
+        }
+
+        private static void DebugMessage(string message)
+        {
+            if (s_debugMode)
+            {
+                Console.WriteLine(message);
+            }
+        }
+
+        private static void PrintUsage()
+            => Console.Error.WriteLine(
+                "Usage:\n ycsb_spanner_csharp run -P <relative_path_to_template_file> [-p property=value]...");
```

Comment:
Unless ycsb_spanner_csharp exists somewhere I'm not aware of, maybe just point at the readme.md file?

Reply:
It's the old name of the command before I renamed it.  I've updated it and added the proper "dotnet ..." prefix.